automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala

```diff
@@ -0,0 +1,211 @@
+            // grant reader access to student A
+            val shareWithStudentA: Set[AclEntry] = Set(AclEntry(studentA.email, WorkspaceAccessLevel.Reader, Some(false), Some(false)))
+            Rawls.workspaces.updateAcl(projectName, workspaceName, shareWithStudentA)(ownerAuthToken)
+
+            val studentAWorkspaceDetails = Rawls.workspaces.getWorkspaceDetails(projectName, workspaceName)(studentAToken)
+            parseWorkspaceAccessLevel(studentAWorkspaceDetails) should equal(WorkspaceAccessLevel.Reader.toString())
+            parseShareAccess(studentAWorkspaceDetails) should be(false)
+
+            // check that student A cannot share the workspace with student B
+            val shareWithStudentB: Set[AclEntry] = Set(AclEntry(studentB.email, WorkspaceAccessLevel.Reader, Some(false), Some(false)))
+            assertThrows[RestException](Rawls.workspaces.updateAcl(projectName, workspaceName, shareWithStudentB)(studentAToken))
+            assertNoAccessToWorkspace(projectName, workspaceName)(studentBToken)
+          }(ownerAuthToken)
+        }
+      }
+
+      "to add writers with can-compute access and then revoke can-compute" in {
+        withCleanBillingProject(owner) { projectName =>
+          withWorkspace(projectName, s"${UUID.randomUUID().toString()}-compute-writer") { workspaceName =>
+            val writerCanCompute: Set[AclEntry] = Set(AclEntry(studentA.email, WorkspaceAccessLevel.Writer, Some(false), Some(true)))
+            Rawls.workspaces.updateAcl(projectName, workspaceName, writerCanCompute)(ownerAuthToken)
+
+            val canComputeWorkspaceDetails = Rawls.workspaces.getWorkspaceDetails(projectName, workspaceName)(studentAToken)
+            parseWorkspaceAccessLevel(canComputeWorkspaceDetails) should equal(WorkspaceAccessLevel.Writer.toString())
+            parseComputeAccess(canComputeWorkspaceDetails) should be(true)
+
+            val revokeCanCompute: Set[AclEntry] = Set(AclEntry(studentA.email, WorkspaceAccessLevel.Writer, Some(false), Some(false)))
+            Rawls.workspaces.updateAcl(projectName, workspaceName, revokeCanCompute)(ownerAuthToken)
+
+            val noComputeWorkspaceDetails = Rawls.workspaces.getWorkspaceDetails(projectName, workspaceName)(studentAToken)
+            parseWorkspaceAccessLevel(noComputeWorkspaceDetails) should equal(WorkspaceAccessLevel.Writer.toString())
+            parseComputeAccess(noComputeWorkspaceDetails) should be(false)
+          }(ownerAuthToken)
+        }
+      }
+
+      "to change a writer's access level to reader" in {
+        withCleanBillingProject(owner) { projectName =>
+          withWorkspace(projectName, s"${UUID.randomUUID().toString()}-reader-to-writer") { workspaceName =>
+            val writerAccess: Set[AclEntry] = Set(AclEntry(studentA.email, WorkspaceAccessLevel.Writer))
+            Rawls.workspaces.updateAcl(projectName, workspaceName, writerAccess)(ownerAuthToken)
```

Comment:
Currently failing because it can't find updateAcl https://fc-jenkins.dsp-techops.broadinstitute.org/job/rawls-fiab-test-runner/10966/console